scripts/fastlane/Fastfile

```diff
@@ -0,0 +1,192 @@
+RELEASE_TAG_PREFIX = 'release_v'
+default_platform(:android)
+
+platform :android do |options|
+  gradle_properties_path = File.expand_path("#{Dir.pwd()}/../../gradle.properties")
+  gradle_project_root = File.dirname(gradle_properties_path)
+  change_log_path = "#{gradle_project_root}/CHANGELOG.md"
+  version_regex = /(\d*)\.(\d*)\.(\d*)(?:-unstable\.)?(\d*)?/
+  pr_body = %Q(
+## Release PR review checklist
+- [ ] Verify version name in gradle.properties
+- [ ] Verify version code in gradle.properties
+- [ ] Verify CHANGELOG.md
+)
+
+  desc "Fetch all tags to ensure accurate calculation of the next release number."
+  before_all do 
+    sh('git', 'fetch', '--tags')
+    sh('git', 'fetch')
+  end
+
+  lane :configure_git_options do |options|
+    sh('git', 'config', 'user.email', options[:git_user_email])
+    sh('git', 'config', 'user.name', options[:git_user_name])
+  end
+
+  lane :manage_next_release_pr do |options|
+    product_name = options[:product_name]
+    last_release_tag = last_git_tag(pattern: "#{RELEASE_TAG_PREFIX}*")
+    next_version, commits = calculate_next_release_version(release_tag_prefix: RELEASE_TAG_PREFIX, from_tag: last_release_tag)
+    UI.message("Version change #{last_release_tag} => #{next_version}")
+    update_gradle_properties(version: next_version)
+    update_change_log(last_release_tag: last_release_tag)
+
+    git_add(path: [gradle_properties_path, change_log_path])
+    git_commit(path: [gradle_properties_path, change_log_path], message: "chore(release): #{product_name} - #{next_version}")
+    push_to_git_remote(force: true)
+
+
+    current_branch = sh('git', 'branch', '--show-current')
+    repo_name_parts = options[:repo].split('/')
+    create_pull_request(base: 'main', 
+                        title: "chore(release): #{product_name} - #{next_version}", 
+                        body:pr_body, 
+                        repo: options[:repo], 
+                        api_token: ENV["RELEASE_MANAGER_TOKEN"])
+  end
+
+
+  desc "Checks out a tag and runs the uploadArchives gradle task."
+  lane :push_tag_to_maven do |options|
+    UI.message("Pushing release #{options[:release_tag]} to maven")
+    sh("git", "checkout", options[:release_tag])
+    publish_to_maven
+  end
+
+  desc "Update changelog"
+  private_lane :update_change_log do |options|
+    changelog_body = changelog_from_git_commits(
```

Comment:
Can we filter the commits so that only commits prefixed with `fix` or `feat` get added to the `CHANGELOG`?

Reply:
It looks like this [particular action](https://docs.fastlane.tools/actions/changelog_from_git_commits/) doesn't have any filtering options beyond including/excluding merge commits

Reply:
Here's a fastlane plugin that looks like it does exactly what we want: https://github.com/xotahal/fastlane-plugin-semantic_release/blob/master/README.md

Reply:
I think I looked at this one before and ran into an issue. Can't remember what it was, but I think it had something to do with customizing the version bump logic. I'll check and see if we can just use it for generating changelog.

Reply:
We could also just write it ourselves, and just use [the source of that plugin](https://github.com/xotahal/fastlane-plugin-semantic_release/blob/master/lib/fastlane/plugin/semantic_release/actions/conventional_changelog.rb#L48-L173) as a reference / example.

Reply:
I was able to use the changelog generated using plugin. The issue I had was that in order to generate the changelog you also have to run `analyze_commits`. That function calculates the next version number using a logic that increments the version number on a per commit basis. So you end up skipping version numbers. Not a huge deal, but it's just not what we've been doing.

So, I was able to keep the version number calculated by our action, but replaced the change log generation code to use the semantic_release plugin.